nupic/algorithms/anomaly.py

```diff
@@ -90,26 +91,26 @@ def __init__(self, slidingWindowSize=None, mode=MODE_PURE, binaryAnomalyThreshol
     @param binaryAnomalyThreshold (optional) - if set [0,1] anomaly score
          will be discretized to 1/0 (1 if >= binaryAnomalyThreshold)
          The transformation is applied after moving average is computed and updated.
+         0.0 means disabled (default)
     """
     self._mode = mode
-    if slidingWindowSize is not None:
+    if slidingWindowSize > 0:
       self._movingAverage = MovingAverage(windowSize=slidingWindowSize)
     else:
       self._movingAverage = None
 
-    if self._mode == Anomaly.MODE_LIKELIHOOD or self._mode == Anomaly.MODE_WEIGHTED:
+    if self._mode == Anomaly.MODE_LIKELIHOOD or \
+       self._mode == Anomaly.MODE_WEIGHTED:
       self._likelihood = AnomalyLikelihood() # probabilistic anomaly
-    if not self._mode in Anomaly._supportedModes:
-      raise ValueError("Invalid anomaly mode; only supported modes are: "
-                       "Anomaly.MODE_PURE, Anomaly.MODE_LIKELIHOOD, "
-                       "Anomaly.MODE_WEIGHTED; you used: %r" % self._mode)
+    assert self._mode in Anomaly._supportedModes, "Invalid anomaly mode; \
```

Comment:
Using `assert` like this isn't a good idea because "The current code generator emits no code for an assert statement when optimization is requested at compile time." See https://docs.python.org/2/reference/simple_stmts.html#the-assert-statement